File: leetcode/0344_reverse_string.cpp
// Copyright (c) 2018 Jacopo Notarstefano

#include <algorithm>
#include <cassert>
#include <string>

using namespace std;

class Solution {
 public:
    string reverseString(string s) {
        string result = s;
        reverse(result.begin(), result.end());
        return result;
    }
};

int main() {
    auto solution = Solution();

    assert("olleh" == solution.reverseString("hello"));
}

File: leetcode/0344_reverse_string.py
# -*- coding: utf-8 -*-


class Solution(object):
    def reverseString(self, s):
        return s[::-1]


if __name__ == '__main__':
    solution = Solution()

    assert 'olleh' == solution.reverseString('hello')
    assert 'amanaP :lanac a ,nalp a ,nam A' == solution.\
        reverseString('A man, a plan, a canal: Panama')
